Design-concepts.md

```diff
@@ -260,6 +264,299 @@ For FuseML to be functional, the following design elements and components are ne
+
+
+### KFServing Extensions
+
+
+### Pipeline Templates
+
+
+
+### Workflow
+
+The workflow assumes the following:
+* FuseML is installed in a kubernetes cluster and the FuseML REST API is exposed and can be accessed from outside the cluster (e.g. from a development workstation)
+* 3rd party tools (MLFlow and KFServing) are installed in the same cluster
+* extensions detailed in previous paragraphs are also installed (e.g. runnables are registered with FuseML)
+
+Given all that, running an MLOps pipeline involves the following:
+
+1. define a pipeline template describing what the automated workflow does and what types of inputs (artifact, parameters) it needs
+2. provide all of the pipeline's inputs (artifacts, parameters). An incompletely defined pipeline (a pipeline that doesn't have values for all its inputs) cannot be executed
+
+FuseML should make no assumption about the order in which these steps need to be executed. The pipeline template may be defined before its inputs, in which case it will be automatically triggered when all inputs become available and then subsequently re-triggered whenever new input versions for artifacts are published. Alternatively, all inputs may be available before the pipeline template is created, in which case the pipeline can start immediately upon creation.
+
+The two segments of the workflow are detailed separately in the next sections and they both involve running the fuseml CLI. As a preparation step, the user must configure how the CLI can contact the FuseML API. Several mechanisms could be used for this, all of which should be supported:
+* environment variable (e.g. `export FUSEML_URL=http://fuseml.10.20.30.nip.io`)
+* command line parameter (e.g. `fuseml --url http://fuseml.10.20.30.nip.io`)
+* configuration file (e.g. `~/.fuseml/config`)
+
+Furthermore, when authentication and authorization features are implmented, user credentials will also need to be provided in a similar fashion.
+
+#### Publishing Codesets
+
+In this part of the workflow, the actor responsible for writing MLFlow code (e.g. data scientist) makes the code available to FuseML in the form of a codeset artifact. Several ways of registering FuseML codesets can be imagined, but the simplest one is by running the fuseml CLI on the same machine where the code is located:
+
+* first, create a project - a concept used to group together several codesets and possible other objects, including pipeline templates, with the purpose of applying other operations to the group as a whole (not yet detailed in this document). On the codeset side of things, this is represented by a git organization used as a parent for all codesets belonging to the project. This operation should be restricted so that only FuseML administrators or delegated roles can execute it, but for the purpose of this exercise, we'll assume everyone can do it: 
+
+  ```
+  fuseml create project --name my-ml-project --description "My first ML project with FuseML"
```

Comment:
Could you use `project create` instead? First, you already used such syntax before, and then it fits better to our API generation...

Reply:
of course